app.rb

```diff
@@ -0,0 +1,165 @@
+    puts '1 - List all books'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals for a given person id'
+    puts '7 - Exit'
+  end
+
+  # rubocop:disable Metrics/CyclomaticComplexity
+  def process_choice(choice)
+    case choice
+    when 1 then list_books
+    when 2 then list_people
+    when 3 then create_person
+    when 4 then create_book
+    when 5 then create_rental
+    when 6 then list_rentals_for_person
+    when 7 then exit_app
+    else
+      puts 'Invalid choice. Please choose a valid option.'
+    end
+  end
+  # rubocop:enable Metrics/CyclomaticComplexity
+
+  def list_books
+    puts 'List of books:'
+    book_list.each_with_index do |book, index|
+      puts "#{index}) Title: #{book.title}, Author: #{book.author}"
+    end
+  end
+
+  def list_people
+    puts 'List of people:'
+    people_list.each_with_index do |person, index|
+      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  def create_person
+    puts 'Do you want to create a student (1) or a teacher (2)? [Input the number 1 or 2]:'
+    person_type = gets.chomp.to_i
+
+    puts 'Age:'
+    age = gets.chomp.to_i
+
+    puts 'Name:'
+    name = gets.chomp
+
+    if person_type == 1
+      create_student(age, name)
+    elsif person_type == 2
+      create_teacher(age, name)
+    else
+      puts 'Invalid choice. Please choose a valid option (1 for student, 2 for teacher).'
+    end
+  end
+
+  def create_student(age, name)
+    puts 'Has parent permission? [Y/N]:'
+    parent_permission_input = gets.chomp.downcase
+    parent_permission = parent_permission_input == 'y'
+
+    student = Student.new(classroom: nil, age: age, name: name, parent_permission: parent_permission)
+    people_list << student
+    puts 'Student created successfully!'
+  end
+
+  def create_teacher(age, name)
+    puts 'Specialization:'
+    specialization = gets.chomp
+
+    teacher = Teacher.new(specialization: specialization, age: age, name: name)
+    people_list << teacher
+    puts 'Teacher created successfully!'
+  end
+
+  def create_book
+    puts 'Title:'
+    title = gets.chomp
+
+    puts 'Author:'
+    author = gets.chomp
+
+    book = Book.new(title, author)
+    book_list << book
+    puts 'Book created successfully!'
+  end
+
+  def create_rental
+    puts 'Select a book from the following list by number:'
+    list_books
+
+    book_index = gets.chomp.to_i
+
+    puts 'Select a person from the following list by number (not id):'
+    list_people
+
+    person_index = gets.chomp.to_i
+
+    puts 'Date (yyyy/mm/dd):'
+    date_str = gets.chomp
+
+    book = book_list[book_index]
+    person = people_list[person_index]
+
+    rental = Rental.new(::Date.strptime(date_str, '%Y/%m/%d'), person, book)
+    rental_list << rental
+
+    puts 'Rental created successfully!'
+  end
```

Comment:
- [ ] As stated in the project requirement Kindly make sure that Your console app should behave in the same way as it is in the example. When I create rental I got an error as shown below with screenshot, The error message you’re seeing is indicating that the `add_rental` method in your `Person class` is expecting two arguments, but only one was provided when the method was called. Here’s the method definition in your Person class:
```
def add_rental(date, book)
  rental = Rental.new(date, self, book)
  @rentals << rental
end

```
![image](https://github.com/mryeminthu/school-library/assets/93598189/cfaaec68-ea82-4ac3-93b8-353f9429807b)

- This method is expecting two arguments: **date and book**. The error is being thrown when this method is called in your `app.rb` file: So the issue. In your `create_rental` method, you’re creating a new Rental instance directly. However, you should be using the `add_rental` method of the Person class to create a new rental and add it to the person’s rentals. So Kindly Fix the issues.